ci: update feature gates setting from minikube.sh

Rakshith-R opened the pull request:
BlockVolume, CSIBlockVolume(GA since k8s v1.18) & VolumeSnapshotDataSource (GA since k8s v1.20) default to true and don't need to be set to true in feature gates setting.
Removing feature gates since some options are removed from newer versions of k8s.

Signed-off-by: Rakshith R <rar@redhat.com>

Reference: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ 

This should also fix e2e fails in #1963 for k8s v1.21.0. 
Updates: #1963 #1968 

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

Rakshith-R commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
> /retest ci/centos/mini-e2e/k8s-1.19

`Error from server: error when creating "https://raw.githubusercontent.com/rook/rook/v1.4.9/cluster/examples/kubernetes/ceph/toolbox.yaml": etcdserver: request timed out`
Rook deployment failed

Yuggupta27 commented on scripts/minikube.sh:
```diff
@@ -216,16 +213,16 @@ up)
         enable_psp
         # shellcheck disable=SC2086
-        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}" --cni="${CNI}" ${EXTRA_CONFIG}
+        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}"  --cni="${CNI}" ${EXTRA_CONFIG}
```
```suggestion
        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --cni="${CNI}" ${EXTRA_CONFIG}
```

Yuggupta27 commented on scripts/minikube.sh:
```diff
@@ -216,16 +213,16 @@ up)
         # This is a workaround to fix psp issues in minikube >1.6.2 and <1.11.0
         # shellcheck disable=SC2086
-        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}" --cni="${CNI}"
+        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}"  --cni="${CNI}"
```
```suggestion
        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --cni="${CNI}"
```

Yuggupta27 commented on scripts/minikube.sh:
```diff
@@ -216,16 +213,16 @@ up)
         ${minikube} stop
         # shellcheck disable=SC2086
-        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}" --cni="${CNI}" ${EXTRA_CONFIG}
+        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}"  --cni="${CNI}" ${EXTRA_CONFIG}
```
```suggestion
        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --cni="${CNI}" ${EXTRA_CONFIG}
```

Yuggupta27 reviewed (changes requested):
You can get rid of extra white spaces.

Rakshith-R commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
> /retest ci/centos/mini-e2e/k8s-1.19

Rook deploy failed

Madhu-1 commented on scripts/minikube.sh:
```diff
@@ -216,16 +213,16 @@ up)
         enable_psp
         # shellcheck disable=SC2086
-        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --feature-gates="${K8S_FEATURE_GATES}" --cni="${CNI}" ${EXTRA_CONFIG}
+        ${minikube} start --force --memory="${MEMORY}" --cpus="${CPUS}" -b kubeadm --kubernetes-version="${KUBE_VERSION}" --driver="${VM_DRIVER}" --cni="${CNI}" ${EXTRA_CONFIG}
```
what if someone wants to pass a newly added feature gate?

Rakshith-R replied:
Got it :+1: ,Made some changes to preserve capability to set feature gates option with default `"ExpandCSIVolumes=true"` (still in beta).

Rakshith-R committed:
ci: update feature gates setting from minikube.sh

BlockVolume, CSIBlockVolume(GA since k8s v1.18) & VolumeSnapshotDataSource
(GA since k8s v1.20) default to true and don't need to be set to true in
feature gates setting.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R commented:
k8s v1.21.0 usage with current minikube.sh fails with following minikube logs. This commit should also fix e2e fails in #1963 .
```
Apr 14 04:16:44 minikube kubelet[4837]: E0414 04:16:44.943714    4837 server.go:186] "Failed to set feature gates from initial flags-based config" err="unrecognized feature gate: CSIBlockVolume"

Apr 14 04:16:44 minikube kubelet[4824]: E0414 04:16:44.195162    4824 server.go:186] "Failed to set feature gates from initial flags-based config" err="unrecognized feature gate: BlockVolume"
```

humblec commented:
@Madhu-1 I feel its good to take this in `v3.3` release , wdyt?

Madhu-1 commented:
> @Madhu-1 I feel its good to take this in `v3.3` release , wdyt?

sure we can take it in

Madhu-1 commented:
> This pull request has been merged with the
> unsupported configuration option `bot_account`.
> 
> This option will be ignored starting May 1st, 2021, and removed
> on June 1st, 2021.
> 
> This option can be replaced by `update_bot_account`, `merge_bot_account` or both
> depending on your use-case (https://docs.mergify.io/actions/merge/).

@Rakshith-R can you fix this bot issue in mergify.yml?
